src/Damage.js

```diff
@@ -25,7 +27,9 @@ class Damage {
       attacker = null,
       type = "physical",
       source = null,
-      hidden = false
+      hidden = false,
+      critical = false,
+      glancing = false
```

Comment:
I don't see you using `glancing` anywhere so perhaps we can remove that for now.